src/file_pipe_log/pipe_builder.rs

```diff
@@ -401,6 +402,18 @@ impl<F: FileSystem> DualPipesBuilder<F> {
 
     /// Builds a new storage for the specified log queue.
     fn build_pipe(&self, queue: LogQueue) -> Result<SinglePipe<F>> {
+        // Here, we would calculate the `Recycle Capacity` based on the relevant
+        // configurations in `cfg`.
+        // Attention please, `capacity_of_recycle` would be dynamically updated
+        // by the formula:
+        // `Config::purge_threshold / Config::targe_file_size - active_file_count`
+        let capacity_of_recycle = |default_file_size: usize, purge_threshold: usize| -> usize {
```

Comment:
Add a configuration for it in config.rs. Set the default value to `purge_threshold`.

We will also do the check on format_version there. i.e. capacity is set to 0 unless format_version>=2.